src/psyclone/psyir/nodes/node.py

```diff
@@ -120,7 +121,6 @@ class Node(object):
     valid_annotations = tuple()
 
     def __init__(self, ast=None, children=None, parent=None, annotations=None):
-        from psyclone.psyGen import InternalError
         if not children:
             self._children = []
         else:
```

Comment:
pylint reports an error at line 342 (undefined `Loop`).

Reply:
I believe it actually needs the following patch:
```
diff --git a/src/psyclone/psyir/nodes/node.py b/src/psyclone/psyir/nodes/node.py
index ef944b45..55af5282 100644
--- a/src/psyclone/psyir/nodes/node.py
+++ b/src/psyclone/psyir/nodes/node.py
@@ -265,6 +265,7 @@ class Node(object):
         try:
             graph = gv.Digraph(format=file_format)
         except ValueError:
+            from psyclone.psyGen import GenerationError
             raise GenerationError(
                 "unsupported graphviz file format '{0}' provided".
                 format(file_format))
@@ -339,6 +340,7 @@ class Node(object):
             graph.edge(remote_name, local_name, color="blue")
         # now call any children so they can add their information to
         # the graph
+        from psyclone.psyir.nodes import Loop
         if isinstance(self, Loop):
             # In case of a loop only loop at the body (the other part
             # of the tree contain start, stop, step values):

```

Reply:
OK, that was stupid - I just copied the patch I had to apply on my branch, without checking if this patch had already added the import of GenerationError ... which it had. So ignore the patch, but Loop is indeed needed :blush:

Reply:
I added the import but the pylint still complain about next line as self(Node).loop_body may not exist as it does not take the condition into consideration, but I think it is OK for now.

Reply:
Thanks - I agree.